pyfstat/core.py

```diff
@@ -1099,27 +1101,19 @@ def plot_twoF_cumulative(
         """
         if ax is None:
             fig, ax = plt.subplots()
-        if injectSources:
-            pfs_input = dict(
-                h0=injectSources["h0"],
-                cosi=injectSources["cosi"],
-                psi=injectSources["psi"],
-                Alpha=injectSources["Alpha"],
-                Delta=injectSources["Delta"],
-                Freq=injectSources["fkdot"][0],
-            )
-        else:
-            pfs_input = None
+            ax.grid()
 
-        taus, twoFs = self.calculate_twoF_cumulative(**kwargs)
+        taus, twoFs = self.calculate_twoF_cumulative(**calculate_twoF_cumulative_kwargs)
         ax.plot(taus / 86400.0, twoFs, label=plt_label, color=c)
         if len(self.detector_names) > 1:
             detector_names = self.detector_names
             detectors = self.detectors
             for d in self.detector_names:
                 self.detectors = d
                 self.init_computefstatistic()
```

Comment:
This doesn't do anything, because `init_computefstatistic()` calls `_get_SFTCatalog` which itself returns immediately without change `if hasattr(self, "SFTCatalog")`. Not sure if

1. that is sane behaviour normally and we should only give those two options an optional override `enforceNewCatalog=False`
1. or we should just remove that early return and assume "whenever somebody calls a re-init, they want to start from scratch"

Probably requires a deeper look at other use cases of "re-initing".

Reply:
If this is difficult to resolve/decide, fallback from above: "drop single-IFO support" (for `plot_twoF_cumulative()`, that is).

Reply:
I'll try to do a single-IFO from within this method; if I don't succeed, I'll just drop it and tell the user "Look, don't give me two IFOs if you only need one: Use constraints".

Reply:
The amount of overhaul required to implement a multi-IFO is huge, and doing so would not be any different from just doing a different ComputeFstat set up (one for each of the IFO constraints).

Reply:
I made #189 to solve this.